File: Algorithm/baekjoon/bj_2630.py
def dnc(n, x=0, y=0):
    global white, blue
    col = arr[x][y]
    for i in range(x,x+n):
        for j in range(y,y+n):
            if col != arr[i][j]:
                t=n//2
                for dx,dy in (0,0),(0,t),(t,0),(t,t) : dnc(t,x+dx,y+dy)
                return
    if col : blue+=1
    else : white+=1

N = int(input()) 
arr = [[*map(int,input().split())] for _ in range(N)]
white,blue = 0,0
dnc(N)

print(white)
print(blue)

File: Algorithm/baekjoon/java/src/bj_2630.java
import java.io.*;
import java.util.*;

public class Main {
    static int white = 0;
    static int blue = 0;
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st ;

        // N : 배열의 크기
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());

        // arr : 색종이 배열
        int[][] arr = new int[N][N];
        for (int i=0; i<N; i++) {
            st = new StringTokenizer(br.readLine());
            for (int j=0; j<N; j++) {
                arr[i][j] = Integer.parseInt(st.nextToken());
            }
        }
        dnc(arr,0,0,N);

        System.out.println(white);
        System.out.println(blue);
    }

    public static void dnc(int[][] arr, int x, int y, int n) {
        if (checkArr(arr,x,y,n)) {
            if (arr[x][y]==0) {
                white ++;
            } else {
                blue ++;
            }
            return;
        }
        int t = n/2;
        dnc(arr,x,y,t);
        dnc(arr,x,y+t,t);
        dnc(arr,x+t,y,t);
        dnc(arr,x+t,y+t,t);
    }

    public static boolean checkArr(int[][]arr,int x,int y,int n) {
        int col = arr[x][y];
        for (int i=x; i<x+n; i++) {
            for (int j=y; j<y+n; j++) {
                if (arr[i][j] != col) {
                    return false;
                }
            }
        }
        return true;
    }
}